### ISSUE TITLE
Model listener firing on changes in unrelated attribute

### ISSUE THREAD

**Issue by casio** [rhysbrettbowen/PlastronJS#17]
Hi Rhys,

the following test - inserted into model_test.js - will run into an infinite loop.

```
var testBindFnWithSetter = function() {
  var arr = 0, curItem = 0, silent = false;
  var m = mvc.Model.create({
    'arr': [1,2,3],
    'curItem': 0
  });
  m.bind('curItem', function() {
    curItem++;
    m.set('arr', [4,5], silent);
  });
  m.bind('arr', function() { arr++; });
  m.set('curItem', 1);
  assertEquals('Should have fired "arr" listener', arr, 1);
  assertEquals('Should have changed "arr"', m.get('arr').length, 2);
  assertEquals('Should have fired "curItem" listener', curItem, 1);
}
```

If you set `silent = true`  you get rid of the infinite loop, but the listener on `arr` of course wont get fired.
To me this seems like a bug. Or is there something wrong in my usage of bind?

**Comment by rhysbrettbowen** [comment]
problem is that when you have a set inside of a bind then another change will be fired and since the previous value of curItem hasn't been updated it sees curItem as a change and will run that again and so on. Best thing to do is set on silent and then manually fire a change on a setTimeout. I was considering running changes asynchronously which should solve the problem but could cause issues if you need values to be reflected at the time. I might add in an asyncBind in the meantime which could just run the bound function asynchronously (using a setTimeout).

**Comment by casio** [comment]
Thanks for the quick response!

Hm, thats interesting. I was thinking bind() would be called only after the actual change did happen.
Users could still refer to prev() to get the prev value - but I just saw that prev is set after dispatching of the change event.

Maybe thats only me, but this would feel a bit more intuitive. However, to not break the api, an asyncBind() would be great already, so users wouldnt have to clutter their code with setTimeout.

Fyi in my usecase, I was still not getting the desired behavior when I did:

```
m.set('curItem', 1, true);  
setTimeout(function() {
  m.change();
}, 0)
```

..what did work was this:

```
setTimeout(function() {
    m.set('curItem', 1, false);
}, 0)
```
